ipset_test.go

```diff
@@ -244,25 +244,45 @@ func TestIPSetIntersection(t *testing.T) {
 		intersection []string
 	}{
 		{
-			[]string{"10.0.0.0/20", "10.5.8.0/24", "10.23.324.100/23"},
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
+			[]string{"10.0.20.0/20", "10.5.8.0/24", "10.23.224.100/23"},
+			[]string{"10.0.20.2/30", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.23.224.104/27", "10.0.20.2/30", "10.5.8.200/29"},
 		},
 		{
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
-			[]string{"10.0.0.0/20", "10.5.8.0/24", "10.23.324.100/23"},
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
+			[]string{"10.10.0.2/30", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.10.0.0/20", "10.5.8.0/24", "10.23.224.100/23"},
+			[]string{"10.10.0.2/30", "10.5.8.200/29", "10.23.224.104/27"},
 		},
 		{
-			[]string{"10.0.5.0/24", "10.5.8.200/29", "10.23.324.104/27"},
-			[]string{"10.6.0.0/24", "10.9.9.0/29", "10.23.6.100/23"},
+			[]string{"10.0.5.0/24", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.6.0.0/24", "10.9.9.1/29", "10.23.6.100/23"},
 			[]string{},
 		},
 		{
-			[]string{"10.23.6.100/24", "10.5.8.200/29", "10.23.324.104/27"},
+			[]string{"10.23.6.100/24", "10.5.8.200/29", "10.23.224.104/27"},
 			[]string{"10.6.0.0/24", "10.9.9.0/29", "10.23.6.200/29"},
 			[]string{"10.23.6.200/29"},
 		},
+		{
+			[]string{"2001:db8:0:23::/96", "2001:db8:0:20::/96", "2001:db8:0:15::/96"},
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "2001:db8:0:15::/64"},
+			[]string{"2001:db8:0:23::/96", "2001:db8:0:20::/96", "2001:db8:0:15::/96"},
+		},
+		{
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "2001:db8:0:15::/64"},
+			[]string{"2001:db8:0:23::/96", "2001:db8:0:20::/96", "2001:db8:0:15::/96"},
+			[]string{"2001:db8:0:15::/96", "2001:db8:0:20::/96", "2001:db8:0:23::/96"},
+		},
+		{
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "2001:db8:0:15::/64"},
+			[]string{"2001:db8:0:14::/96", "2001:db8:0:10::/96", "2001:db8:0:8::/96"},
+			[]string{},
+		},
+		{
+			[]string{"2001:db8:0:23::/64", "2001:db8:0:20::/64", "172.16.1.45/24"},
+			[]string{"2001:db9:0:14::/96", "2001:db9:0:10::/96", "172.16.1.200/28"},
+			[]string{"172.16.1.200/28"},
```

Comment:
Thanks for adding all of the ipv6 tests. IPv6 and IPv4 should be given the same level of priority in this library as a whole.